core/DataAccess/Model.php

```diff
@@ -115,10 +115,17 @@ public function updateArchiveAsInvalidated($archiveTable, $idSites, $datesByPeri
         foreach ($datesByPeriodType as $periodType => $dates) {
             $dateConditions = array();
 
-            foreach ($dates as $date) {
-                $dateConditions[] = "(date1 <= ? AND ? <= date2)";
-                $bind[] = $date;
-                $bind[] = $date;
+            if ($periodType == Period\Range::PERIOD_ID) {
```

Comment:
This is the `if` I mean below. I reckon we could restore old behaviour and delete all ranges it affects when using `--period=range`. I understand when using `--period=range` the problem is that it wouldn't iterate over all archive tables. For this we could always call `markArchivesOverlappingRangeAsInvalidated()` for the ranges and instead of `updateAllRangeArchivesAsInvalidated` call `updateArchiveAsInvalidated`.